vmware-event-router/internal/processor/processor.go

```diff
@@ -11,3 +13,21 @@ import (
 type Processor interface {
 	Process(types.ManagedObjectReference, []types.BaseEvent) error
 }
+
+// Error struct contains the generic error content used by the processors
+// it extends the simple error by providing context which processor gave
+// the error
+type Error struct {
+	processor string
+	err       error
+}
+
+// NewError constructs Error structure
+func NewError(processor string, err error) *Error {
```

Comment:
Do we need to export this one? 
Also: what do you think of renaming this to:

```go
func ProcessorError(processor string, err error) error {}
```

This would then be used via `return ProcessorError(ProviderAWS, errMsg)`

Reply:
Well it is in the same package, if we are going to use the error in the processors only there is no need to export it. It also looks prettier that way 😄 will lowercase it so it is not exported outside the package. Otherwise will change as proposed 👍

Reply:
Internal for the package ✔️